### DISCUSSION TITLE
Is GitHub Pages (or GitLab Pages) required for publication?

### DISCUSSION THREAD

**Discussion by jorge-campo** [squidfunk/mkdocs-material#3735] [Q&A] [answered]
Hi! Maybe this is a stupid question, but I've checked the Material for MkDocs [publishing help](https://squidfunk.github.io/mkdocs-material/publishing-your-site/) and reviewed other Q&As here, and I can't find the answer to this one.

My organization is moving away from Cloudfare, so GitHub Pages is not an option. Is it possible to publish a Material for MkDocs site without GitHub pages (or GitLab Pages)? The Material documentation [discusses this option](https://www.mkdocs.org/user-guide/deploying-your-docs/), but I would like to be sure this also applies to MkDocs as well.

I'm aware CI/CD is unavoidable as per https://github.com/mkdocs/mkdocs/discussions/2599.

**Comment by facelessuser** [comment, marked as answer]
No, github pages and gitlab pages are not required to use Material, the entire documentation site is created using MkDocs and should work wherever you push it. `mkdocs gh-deploy` just makes it easy to push to things like GitHub.

**Reply by squidfunk** [reply]
@facelessuser is right. After invoking `mkdocs build` you can push the contents of the `site` directory wherever you like.

**Reply by jorge-campo** [reply]
Thank you! And thanks for making this fantastic product.

Currently, we are finishing up our testing, and if everything goes as expected, we would be more than happy to sponsor Material.
